src/AppBundle/Controller/DonationController.php

```diff
@@ -91,11 +92,15 @@ public function callbackAction(Request $request)
     {
         $id = explode('_', $request->query->get('id'))[0];
```

Comment:
This line and the following condition should be encapsulated somewhere in a static method. Something similar to this.

```
try {
    Donation::validateId($request->query->get('id'));
} catch (InvalidDonationIdException $e) {
    return $this->redirectToRoute('donation_index');
}